src/ray/gcs/entry_change_notification.h

```diff
@@ -0,0 +1,51 @@
+#ifndef RAY_GCS_ENTRY_CHANGE_NOTIFICATION_H
+#define RAY_GCS_ENTRY_CHANGE_NOTIFICATION_H
+
+#include <ray/protobuf/gcs.pb.h>
+#include <vector>
+
+namespace ray {
+
+namespace gcs {
+
+/// \class EntryChangeNotification
+/// EntryChangeNotification class is a template class which represent
+/// notification of entry change from GCS.
+template <typename Data>
+class EntryChangeNotification {
+ public:
+  EntryChangeNotification(rpc::GcsChangeMode change_mode, std::vector<Data> data)
```

Comment:
Should data be a single item, instead of a vector?
If we have multiple items to notify changes, we should use multiple `EntryChangeNotification` objects.